docs/rackhd/event_notification.rst

```diff
@@ -1,105 +1,272 @@
 Event Notification
 ------------------
 
-The change of resources managed by RackHD could be retrieved from AMQP messages.
-
-- Exchange: **on.events**
-- Routing Key **event.<type>**
+RackHD support event notification via both AMQP and web hook.
 
-`<type>` is event's type, it's the same with event's payload `type`, see payloads_ for more details.
+A web hook allows an application to subscribe certain RackHD published events by configured URL, when one of the subscribed events is triggered, RackHD will send a HTTP POST request with event payload to configured URL.
 
-An example of listening node events using tool 'sniff.js' located at https://github.com/RackHD/on-tools/blob/master/dev_tools/README.md.
+RackHD also publishes defined events over AMQP, so subscribers to RackHD's instance of AMQP don't need to register a webhook URL to get events. The AMQP events can be prolific, so we recommend that consumers filter events as they are received to what is desired.
 
-.. code-block:: Bash
-
-    $ sudo node sniff.js "on.events" "event.#"
+Events Payloads
+~~~~~~~~~~~~~~~~~~~~
 
-.. _payloads:
+.. _common_event_payload_format:
 
-`type` and `action` fields exist for all types of events.
+All published external events' payload formats are common, the event attributes are as below:
 
 ========= ====== =================================
 Attribute Type   Description
 ========= ====== =================================
-type      String The event type, it could be `node`, `sku`, etc.
-action    String The `action` that was performed, it's related to event `type`
+version   String Event payload format version.
+type      String It could be one of the values: **heartbeat**, **node**, **polleralert**, **graph**.
+action    String a verb or a composition of component and verb which indicates what happened, it's performed by the `type` attribute.
+severity  String Event severity, it could be one of the values: **critical**, **warning**, **information**.
+typeId    String It could be graph's Id, poller's Id, or service name, or filename, etc. it's performed by the `type` attribute.
```

Comment:
Suggest changing the description of typeId:
```
Event type identifier. It varies with even type, it could be node id for node event, graph id for graph event, poller id for poller event, .etc.
```